### PR TITLE
fix(web-core): JSON-stringify objects/arrays in formatString per spec

### LINKED ISSUE #1386: [web_core] formatString should JSON-stringify objects/arrays per spec
`web_core`'s `FormatStringImplementation` (`renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.ts`) uses `Array.prototype.join('')`, which falls back to JS's default `String(value)` for object/array values. This contradicts `a2ui_protocol.md` §"Type conversion":

> **Objects/Arrays**: Stringified as JSON to ensure consistency across different client implementations.

Related but distinct from #912, which only standardized scalar (`null`/`undefined`/`Number`/`Boolean`) coercion.

## Reproduction

Given a data model:

```json
{
  "tags":   ["swift", "ios"],
  "user":   { "name": "Alice", "age": 30 },
  "matrix": [[1, 2], [3, 4]],
  "vals":   [1, null, 3]
}
```

| Template           | Spec says                         | web_core actually outputs               |
| ------------------ | --------------------------------- | --------------------------------------- |
| `"Tags: ${/tags}"` | `Tags: ["swift","ios"]`           | `Tags: swift,ios`                       |
| `"User: ${/user}"` | `User: {"name":"Alice","age":30}` | `User: [object Object]`                 |
| `"M = ${/matrix}"` | `M = [[1,2],[3,4]]`               | `M = 1,2,3,4` (nested arrays flattened) |
| `"V = ${/vals}"`   | `V = [1,null,3]`                  | `V = 1,,3` (null silently dropped)      |

The object case is the most damaging: any object value renders as the literal string `[object Object]`, which is essentially data loss.

## Impact

Affects `web_core` and every renderer that re-exports `BASIC_FUNCTIONS` from it (`react`, `lit`, `angular`).

### PR DESCRIPTION
**Pull request by BBC6BAE9**
# Description

`FormatStringImplementation` used `Array.prototype.join('')` which relied on JS default `String(value)` coercion, producing `[object Object]` for objects and flattened comma-separated values for arrays. The `a2ui_protocol.md` §"Type conversion" requires objects/arrays to be stringified as JSON.

**Changes:**
- Add `coerceToString` helper that implements the spec's type conversion rules: `null`/`undefined` → `""`, objects/arrays → `JSON.stringify()`, scalars → `String()`
- Update `FormatStringImplementation` to use `coerceToString`
- Add 5 unit tests covering object, array, nested array, array-with-null, and null interpolation
- Update CHANGELOG

**Reproduction (from issue):**

| Template | Spec says | `web_core` before fix |
|---|---|---|
| `"Tags: ${/tags}"` | `Tags: ["swift","ios"]` | `Tags: swift,ios` |
| `"User: ${/user}"` | `User: {"name":"Alice","age":30}` | `User: [object Object]` |
| `"M = ${/matrix}"` | `M = [[1,2],[3,4]]` | `M = 1,2,3,4` |
| `"V = ${/vals}"` | `V = [1,null,3]` | `V = 1,,3` |

Fixes #1386

## Pre-launch Checklist

- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I read the [Style Guide].
- [x] I have added updates to the [CHANGELOG].
- [x] I updated/added relevant documentation.
- [x] My code changes (if any) have tests.
- [ ] If my branch is on fork, I have verified that [scripts/e2e_test.sh](../scripts/e2e_test.sh) passes.

If you need help, consider asking for advice on the [discussion board].

[CHANGELOG]: ../CHANGELOG.md
[CLA]: https://cla.developers.google.com/
[Contributors Guide]: ../CONTRIBUTING.md
[discussion board]: https://github.com/google/A2UI/discussions
[Style Guide]: ../CONTRIBUTING.md#coding-style

### CODE REVIEW

File: renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.ts [hunk 1]
```diff
@@ -233,6 +233,18 @@ export const EmailImplementation = createFunctionImplementation(EmailApi, args =
...
 // Formatting
+/**
+ * Coerces a value to a string following the a2ui_protocol.md §"Type conversion" rules:
+ * - Numbers/Booleans: Standard string representation.
+ * - null/undefined: An empty string "".
+ * - Objects/Arrays: Stringified as JSON.
+ */
+function coerceToString(value: unknown): string {
+  if (value === null || value === undefined) return '';
+  if (typeof value === 'object') return JSON.stringify(value);
```
[the remark this answers is not in the document]
Comment (ditman) [reply renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.ts]: This is not a bad suggestion @BBC6BAE9, can you add it?

[the remark this answers is not in the document]
Comment (BBC6BAE9) [reply renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.ts]: Done.

File: renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts [hunk 2]
```diff
@@ -299,6 +299,121 @@ describe('BASIC_FUNCTIONS', () => {
...
 
+    it('formatString (object value is JSON-stringified)', (_, done) => {
+      const objModel = new DataModel({user: {name: 'Alice', age: 30}});
+      const objContext = createTestDataContext(objModel, '/');
+
+      const result = invoke(
+        'formatString',
+        {value: 'User: ${user}'},
+        objContext,
+      ) as import('@preact/signals-core').Signal<string>;
```
Comment (josemontespg) [inline renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts]: Can this be imported at the top of the file along with the other imports?

File: renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts [hunk 3]
```diff
@@ -299,6 +299,121 @@ describe('BASIC_FUNCTIONS', () => {
...
+
+      let cleanup: (() => void) | undefined;
+      // eslint-disable-next-line prefer-const
+      cleanup = effect(() => {
+        const val = result.value;
+        if (val) {
```
Comment (josemontespg) [inline renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts]: If value is falsy for some reason, this test would fail with a timeout (I think). I know this pattern is already present in the file, but for these new tests, could we remove the if here?

In case the value is falsy, I think the test should fail based on the value, to provide a better cause in the error message.

**Comment by ditman** [comment]
I can tweak @josemontespg's suggestions and land this. Thanks @BBC6BAE9 for your contribution! Keep them coming!

File: renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts (hunk 2 above)
Comment (ditman) [reply renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts]: Done. This is following the already existing pattern in the file (it also caught my eye), but I don't understand *why*. This might have been there from when signals were introduced to prevent clashes or something? Unsure.

File: renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts (hunk 3 above)
Comment (ditman) [reply renderers/web_core/src/v0_9/basic_catalog/functions/basic_functions.test.ts]: Done, I removed the val guard from everywhere in the tests

**Review by ditman — APPROVED** [review]
LGTM! I've addressed José's comments, and we can land this. Thanks for your contribution @BBC6BAE9!
